coriolis/conductor/rpc/server.py

```diff
@@ -915,6 +941,23 @@ def _update_volumes_info_for_migration_parent_replica(
             self._update_replica_volumes_info(
                 ctxt, replica_id, instance, updated_task_info)
 
+    def _update_volumes_info(
+            self, ctxt, instance, execution, volumes_info):
+        if isinstance(execution.action, models.Migration):
```

Comment:
Checking the operation we're on is bad and we should never have to do it. (neither using `isinstance`, or the type field you add to executions in #58)
We're already being cheeky with what we do in `_handle_post_task_actions`, and it should stay that way IMO.

In this sense, please define separate task classes for the snapshot creation/deletion during Replica executions (and also rename the old tasks please to have `Migration` in the name), and just handle each task normally in `_handle_post_task_actions ` as we currently do.
The actual calls to the providers can stay the same so you'll likely need to duplicate most of the code in the body of the task classes.